documentation/asciidoc/computers/camera/rpicam_apps_building.adoc

```diff
@@ -1,172 +1,256 @@
 ----
 
-To complete the `libcamera` build, use:
+NOTE: You can disable the `gstreamer` plugin by replacing `-Dgstreamer=enabled` with `-Dgstreamer=disabled` during the `meson` build configuration. If you disable `gstreamer`, there is no need to install the `libglib2.0-dev` and `libgstreamer-plugins-base1.0-dev` dependencies.
+
+Now, you can build `libcamera` with `ninja`:
 
+[source,console]
 ----
-ninja -C build   # use -j 2 on Raspberry Pi 3 or earlier devices
-sudo ninja -C build install
+$ ninja -C build
 ----
 
-NOTE: At the time of writing, `libcamera` does not yet have a stable binary interface. Therefore, if you have rebuilt `libcamera`, we recommend continuing and rebuilding `rpicam-apps` from scratch too.
+Finally, run the following command to install your freshly-built `libcamera` binary:
 
-==== Building `libepoxy`
+[source,console]
+----
+$ sudo ninja -C build install
+----
 
-Rebuilding `libepoxy` should not normally be necessary as this library changes only very rarely. If you do want to build it from scratch, however, please follow the instructions below.
+TIP: On devices with 1GB of memory or less, the build may exceed available memory. Append the `-j 1` flag to `ninja` commands to limit the build to a single process. This should prevent the build from exceeding available memory on devices like the Raspberry Pi Zero and the Raspberry Pi 3.
 
-Start by installing the necessary dependencies.
+`libcamera` does not yet have a stable binary interface. Always build `rpicam-apps` after you build `libcamera`.
+
+==== Building `rpicam-apps`
 
+First fetch the necessary dependencies for `rpicam-apps`.
+
+[source,console]
 ----
-sudo apt install -y libegl1-mesa-dev
+$ sudo apt install -y cmake libboost-program-options-dev libdrm-dev libexif-dev
+$ sudo apt install -y meson ninja-build
 ----
 
-Next, check out and build `libepoxy`.
+Download a local copy of Raspberry Pi's fork of `rpicam-apps` from GitHub:
 
+[source,console]
 ----
-cd
-git clone https://github.com/anholt/libepoxy.git
-cd libepoxy
-mkdir _build
-cd _build
-meson
-ninja
-sudo ninja install
+$ git clone https://github.com/raspberrypi/rpicam-apps.git
 ----
 
-==== Building `rpicam-apps`
+Navigate into the root directory of the repository:
 
-First fetch the necessary dependencies for `rpicam-apps`.
+[source,console]
+----
+$ cd rpicam-apps
+----
+
+For desktop-based operating systems like Raspberry Pi OS, configure the `rpicam-apps` build with the following `meson` command:
 
+[source,console]
 ----
-sudo apt install -y cmake libboost-program-options-dev libdrm-dev libexif-dev
-sudo apt install -y meson ninja-build
+$ meson setup build -Denable_libav=true -Denable_drm=true -Denable_egl=true -Denable_qt=true -Denable_opencv=false -Denable_tflite=false
 ----
 
-The `rpicam-apps` build process begins with the following:
+For headless operating systems like Raspberry Pi OS Lite, configure the `rpicam-apps` build with the following `meson` command:
```

Comment:
Only a minor nitpick, but this "For desktop-based operating systems like Raspberry Pi OS..." and "For headless operating systems like Raspberry Pi OS Lite..." wording is slightly inconsistent with the earlier instructions, where it basically said "Do _this_, but if you're running Raspberry Pi OS Lite do _this_ instead". :man_shrugging: